perllib/FixMyStreet/Roles/Translatable.pm

```diff
@@ -18,9 +25,16 @@ sub translate_column {
 sub _translate {
     my ($self, $col, $fallback) = @_;
 
+    my %cols = $self->get_columns;
+    return $cols{msgstr} if $cols{msgstr};
+
     my $schema = $self->result_source->schema;
     my $table = lc $self->result_source->source_name;
     my $id = $self->id;
+    my $lang = $schema->lang || '';
+
+    my $translated = $self->translated->{$table}{$id}{$col}{$lang};
+    return $translated if $translated;
```

Comment:
Doesn't that slow it down? I guess it could store the problem lookup against the original table/id as well, if that's what you mean.